### PR TITLE
Refactor: SNR & SI_SNR

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Unify the notation and keep more descriptive functional names
part of #729

## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/references/modules.rst [hunk 1]
```diff
@@ -97,10 +97,10 @@ SI_SDR
 .. autoclass:: torchmetrics.SI_SDR
     :noindex:
 
-SI_SNR
-~~~~~~
+ScaleInvariantSNR
```
Comment (justusschock) [inline docs/source/references/modules.rst:100]: This introduces a new inconsistency since in the functional part you don't abbreviate snr, let's also be consistent here

Comment (Borda) [reply docs/source/references/modules.rst:100]: to be clear you suggest naming it `ScaleInvariantSignalNoiseRatio`, correct?

Comment (justusschock) [reply docs/source/references/modules.rst:100]: yes either that or abbreviating snr in the functional interface as well :)

Comment (Borda) [reply docs/source/references/modules.rst:100]: on one hand, we have `MeanAbsolutePercentageError` instead of just MAPE but then we have `ROC` instead of  Receiver Operating Characteristic

Comment (quancs) [reply docs/source/references/modules.rst:100]: Cannot we have full name in all metrics levels and have full name and abbreviate name in each small field?

Comment (stancld) [reply docs/source/references/modules.rst:100]: I think the expanded names are beneficial in many cases. However, for example, `BLEU = bilingual evaluation understudy` or `ROUGE =  Recall-Oriented Understudy for Gisting Evaluation` would be really confusing.

Comment (Borda) [reply docs/source/references/modules.rst:100]: @justusschock I found another interesting mix `pit_permutate`

**Comment by quancs** [comment]
And, SNR: signal to noise ratio, not signal noise ratio
same to SDR: signal to distortion ratio

File: docs/source/references/modules.rst (hunk 1 above)
Comment (lucadiliello) [reply docs/source/references/modules.rst:100]: I think that unrolling retrieval names will just add complexity...
- `RetrievalMAP` -> `RetrievalMeanAveragePrecision`
- `RetrievalMRR` -> `RetrievalMeanAveragePrecision`
- `RetrievalNDCG` -> `RetrievalNormalizedDiscountedCumulativeGain`

Comment (Borda) [reply docs/source/references/modules.rst:100]: @quancs  if I am not wrong SDR and SNR are general metrics, so not only for audio, correct?
if so can your implementation can be used also for Image...? :rabbit:

Comment (quancs) [reply docs/source/references/modules.rst:100]: Maybe yes. Images doesn't have the time dim. If that is a problem, maybe we can remove SDR and SNR from the top level.

Comment (Borda) [reply docs/source/references/modules.rst:100]: I do not mind keeping it in top-level, rather thinking about `AudioSDR` and `AudioSND`?

**Comment by Borda** [comment]
@PyTorchLightning/core-metrics updated and keeping abbreviations, pls check it now :rabbit:

**Comment by quancs** [comment]
@Borda Maybe we can consider to put sisdr and sisnr in one file, because they have very limited difference (or to say sisnr is a special alias for sisdr)

File: docs/source/references/functional.rst [hunk 2]
```diff
@@ -38,17 +38,17 @@ si_sdr [func]
...
 
-.. autofunction:: torchmetrics.functional.si_snr
+.. autofunction:: torchmetrics.functional.sisnr
     :noindex:
 
 
-snr [func]
-~~~~~~~~~~
+signal_noise_ratio [func]
+~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (quancs) [inline docs/source/references/functional.rst:49]:
```suggestion
snr [func]
~~~~~~
```
snr right?

File: docs/source/references/functional.rst [hunk 3]
```diff
@@ -38,17 +38,17 @@ si_sdr [func]
...
 
-.. autofunction:: torchmetrics.functional.snr
+.. autofunction:: torchmetrics.functional.signal_noise_ratio
```
Comment (quancs) [inline docs/source/references/functional.rst:51]:
```suggestion
.. autofunction:: torchmetrics.functional.snr
```

**Comment by Borda** [comment]
> @Borda Maybe we can consider to put sisdr and sisnr in one file, because they have very limited difference (or to say sisnr is a special alias for sisdr)

I put sisnr in snr as it seems to be related, right?

File: torchmetrics/functional/audio/snr.py [hunk 4]
```diff
@@ -64,3 +65,31 @@ def snr(preds: Tensor, target: Tensor, zero_mean: bool = False) -> Tensor:
...
+        >>> preds = torch.tensor([2.5, 0.0, 2.0, 8.0])
+        >>> sisnr(preds, target)
+        tensor(15.0918)
+
+    References:
+        [1] Y. Luo and N. Mesgarani, "TaSNet: Time-Domain Audio Separation Network for Real-Time, Single-Channel Speech
+        Separation," 2018 IEEE International Conference on Acoustics, Speech and Signal Processing (ICASSP), 2018, pp.
+        696-700, doi: 10.1109/ICASSP.2018.8462116.
+    """
+    return si_sdr(target=target, preds=preds, zero_mean=True)
```
Comment (quancs) [inline torchmetrics/functional/audio/snr.py:95]: maybe we can consider move this to a new file `sisdr.py`, which has `sisdr` and `sisnr`

Comment (Borda) [reply torchmetrics/functional/audio/snr.py:95]: yes, after the other one is merged

**Comment by Borda** [comment]
waiting on #711

**Comment by quancs** [comment]
> > @Borda Maybe we can consider to put sisdr and sisnr in one file, because they have very limited difference (or to say sisnr is a special alias for sisdr)
> 
> I put sisnr in snr as it seems to be related, right?

Yes, `sisnr` and `snr` sounds more related. I'm wondering which is better: 1 put `sisnr` and `snr` together; 2 put `sisnr` and `sisdr` together; 3 put `sisnr`, `sisdr`, `snr` separately as before

**Comment by Borda** [comment]
> > > @Borda Maybe we can consider to put sisdr and sisnr in one file, because they have very limited difference (or to say sisnr is a special alias for sisdr)
> > 
> > 
> > I put sisnr in snr as it seems to be related, right?
> 
> Yes, `sisnr` and `snr` sounds more related. I'm wondering which is better: 1 put `sisnr` and `snr` together; 2 put `sisnr` and `sisdr` together; 3 put `sisnr`, `sisdr`, `snr` separately as before

I would go with 2

File: docs/source/references/modules.rst (hunk 1 above)
Comment (quancs) [reply docs/source/references/modules.rst:100]: If let me choose, I would prefer remove SDR and SNR from the top level

**Review by quancs — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: torchmetrics/audio/si_snr.py [hunk 5]
```diff
@@ -12,90 +12,35 @@
...
         process_group: Optional[Any] = None,
         dist_sync_fn: Optional[Callable[[Tensor], Tensor]] = None,
     ) -> None:
-        super().__init__(
-            compute_on_step=compute_on_step,
-            dist_sync_on_step=dist_sync_on_step,
-            process_group=process_group,
-            dist_sync_fn=dist_sync_fn,
+        warn(
+            "`SI_SDR` was renamed to `ScaleInvariantSignalNoiseRatio` in v0.7 and it will be removed in v0.8",
```
Comment (Borda) [inline torchmetrics/audio/si_snr.py:43]:
```suggestion
            "`SI_SNR` was renamed to `ScaleInvariantSignalNoiseRatio` in v0.7 and it will be removed in v0.8",
```

File: torchmetrics/audio/snr.py [hunk 6]
```diff
@@ -109,3 +109,111 @@ def update(self, preds: Tensor, target: Tensor) -> None:  # type: ignore
...
+    def __init__(
+        self,
+        zero_mean: bool = False,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Optional[Callable[[Tensor], Tensor]] = None,
+    ) -> None:
+        warn("`SNR` was renamed to `SignalNoiseRatio` in v0.7 and it will be removed in v0.8", DeprecationWarning)
+
```
Comment (Borda) [inline torchmetrics/audio/snr.py:137]:
```suggestion
```
